### ISSUE TITLE
tox -e apacheconftest fails on Ubuntu 18.04+

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#6706]
`tox -e apacheconftest` fails on Ubuntu 18.04+. In addition to possibly being a bug in the Apache plugin which we should look into, it's annoying because it is now the only test farm tests we run that are failing on the AMIs I want to add in #6700.

**WARNING:** For anyone who works on this, these tests make changes to both the packages installed on your system and your Apache configuration.

Both Ubuntu 18.04 and Ubuntu 18.10 failed on the same configuration. The output from one of these runs is:
```
# venv/bin/tox -e apacheconftest
apacheconftest create: /home/ubuntu/letsencrypt/.tox/apacheconftest
apacheconftest installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
apacheconftest runtests: PYTHONHASHSEED='0'
apacheconftest runtests: commands[0] | python /home/ubuntu/letsencrypt/tools/pip_install_editable.py acme . certbot-apache certbot-compatibility-test
"/home/ubuntu/letsencrypt/.tox/apacheconftest/bin/python" -m pip install --constraint "/tmp/tmp1gQACN/all_constraints.txt" -e acme -e . -e certbot-apache -e certbot-compatibility-test
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Obtaining file:///home/ubuntu/letsencrypt/acme
Obtaining file:///home/ubuntu/letsencrypt
Obtaining file:///home/ubuntu/letsencrypt/certbot-apache
Obtaining file:///home/ubuntu/letsencrypt/certbot-compatibility-test
Collecting ConfigArgParse==0.12.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 2))
Collecting configobj==5.0.6 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 21))
Collecting cryptography==2.2.2 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 23))
  Using cached https://files.pythonhosted.org/packages/dd/c2/3a5bfefb25690725824ade71e6b65449f0a9f4b29702cce10560f786ebf6/cryptography-2.2.2-cp27-cp27mu-manylinux1_x86_64.whl
Collecting enum34==1.1.2 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 28))
Collecting idna==2.5 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 35))
  Using cached https://files.pythonhosted.org/packages/11/7d/9bbbd7bb35f34b0169542487d2a8859e44306bb2e6a4455d491800a5621f/idna-2.5-py2.py3-none-any.whl
Collecting ipaddress==1.0.16 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 37))
  Using cached https://files.pythonhosted.org/packages/23/6a/813ac29a01e4c33c19c2bded98ac3d4266ebbf0bd2c0eb0020e1c969958d/ipaddress-1.0.16-py27-none-any.whl
Collecting josepy==1.1.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 42))
  Using cached https://files.pythonhosted.org/packages/23/46/30ac29742b03be92e05d1fd9048df6f0902bb0b3b042a7ea76a2b3c30f7e/josepy-1.1.0-py2.py3-none-any.whl
Collecting mock==1.3.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 46))
  Using cached https://files.pythonhosted.org/packages/b2/50/664a70b87408bb6c14c1af2337efa64eb8d1af80c933531758b8fb41ec25/mock-1.3.0-py2.py3-none-any.whl
Collecting parsedatetime==2.1 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 52))
  Using cached https://files.pythonhosted.org/packages/85/1f/13fc06097e516f6259d62cea502b116451321c96e18a9d0fff9da3442e02/parsedatetime-2.1-py2-none-any.whl
Collecting pbr==1.8.1 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 54))
  Using cached https://files.pythonhosted.org/packages/fc/37/94af8387babb09796d306b18cf94ee5c70388c875a16d8a88e471500452c/pbr-1.8.1-py2.py3-none-any.whl
Collecting pyOpenSSL==18.0.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 62))
  Using cached https://files.pythonhosted.org/packages/96/af/9d29e6bd40823061aea2e0574ccb2fcf72bfd6130ce53d32773ec375458c/pyOpenSSL-18.0.0-py2.py3-none-any.whl
Collecting pyRFC3339==1.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 63))
  Using cached https://files.pythonhosted.org/packages/9b/0a/decfa17e7707afca17d6e9595ff5c79c1c71c74063ad95576f897ed3a9f1/pyRFC3339-1.0-py2.py3-none-any.whl
Collecting python-augeas==0.5.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 73))
Collecting pytz==2015.7 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 76))
  Using cached https://files.pythonhosted.org/packages/c0/28/973f0382c803b21734cd7e97e0590928148ee21b1cbe8f7fed8b506204fb/pytz-2015.7-py2.py3-none-any.whl
Collecting requests==2.20.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 78))
  Using cached https://files.pythonhosted.org/packages/f1/ca/10332a30cb25b627192b4ea272c351bce3ca1091e541245cccbace6051d8/requests-2.20.0-py2.py3-none-any.whl
Collecting requests-toolbelt==0.8.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 80))
  Using cached https://files.pythonhosted.org/packages/97/8a/d710f792d6f6ecc089c5e55b66e66c3f2f35516a1ede5a8f54c13350ffb0/requests_toolbelt-0.8.0-py2.py3-none-any.whl
Collecting six==1.10.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 85))
  Using cached https://files.pythonhosted.org/packages/c8/0a/b6723e1bc4c516cb687841499455a8505b44607ab535be01091c0f24f079/six-1.10.0-py2.py3-none-any.whl
Collecting urllib3==1.24.1 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 99))
  Using cached https://files.pythonhosted.org/packages/62/00/ee1d7de624db8ba7090d1226aebefab96a2c71cd5cfa7629d6ad3f61b79e/urllib3-1.24.1-py2.py3-none-any.whl
Collecting zope.component==4.2.2 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 102))
Collecting zope.event==4.1.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 103))
Collecting zope.interface==4.1.3 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 104))
Requirement already satisfied: setuptools in ./.tox/apacheconftest/lib/python2.7/site-packages (from acme==0.31.0.dev0) (40.6.3)
Collecting cffi==1.11.5 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 18))
  Using cached https://files.pythonhosted.org/packages/14/dd/3e7a1e1280e7d767bd3fa15791759c91ec19058ebe31217fe66f3e9a8c49/cffi-1.11.5-cp27-cp27mu-manylinux1_x86_64.whl
Collecting asn1crypto==0.22.0 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 11))
  Using cached https://files.pythonhosted.org/packages/97/ba/7e8117d8efcee589f4d96dd2b2eb1d997f96d27d214cf2b7134ad8acf6ab/asn1crypto-0.22.0-py2.py3-none-any.whl
Collecting funcsigs==1.0.2 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 30))
  Using cached https://files.pythonhosted.org/packages/69/cb/f5be453359271714c01b9bd06126eaf2e368f1fddfff30818754b5ac2328/funcsigs-1.0.2-py2.py3-none-any.whl
Collecting chardet==3.0.2 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 19))
  Using cached https://files.pythonhosted.org/packages/b8/74/54fdc2fcfdd46b6c486964b64c5bb7db9a3664033ab25cf11aab06dd2a5d/chardet-3.0.2-py2.py3-none-any.whl
Collecting certifi==2017.4.17 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 17))
  Using cached https://files.pythonhosted.org/packages/eb/01/c1f58987b777d6c4ec535b4e004a4a07bfc9db06f0c7533367ca6da8f2a6/certifi-2017.4.17-py2.py3-none-any.whl
Collecting pycparser==2.14 (from -c /tmp/tmp1gQACN/all_constraints.txt (line 66))
Installing collected packages: ConfigArgParse, asn1crypto, certifi, pycparser, cffi, chardet, six, configobj, enum34, idna, ipaddress, cryptography, funcsigs, pyOpenSSL, josepy, pbr, mock, parsedatetime, pytz, pyRFC3339, python-augeas, urllib3, requests, requests-toolbelt, zope.interface, zope.event, zope.component, acme, certbot, certbot-apache, certbot-compatibility-test
  Running setup.py develop for acme
  Found existing installation: certbot 0.31.0.dev0
    Not uninstalling certbot at /home/ubuntu/letsencrypt, outside environment /home/ubuntu/letsencrypt/.tox/apacheconftest
    Can't uninstall 'certbot'. No files were found to uninstall.
  Running setup.py develop for certbot
  Running setup.py develop for certbot-apache
  Running setup.py develop for certbot-compatibility-test
Successfully installed ConfigArgParse-0.12.0 acme asn1crypto-0.22.0 certbot certbot-apache certbot-compatibility-test certifi-2017.4.17 cffi-1.11.5 chardet-3.0.2 configobj-5.0.6 cryptography-2.2.2 enum34-1.1.2 funcsigs-1.0.2 idna-2.5 ipaddress-1.0.16 josepy-1.1.0 mock-1.3.0 parsedatetime-2.1 pbr-1.8.1 pyOpenSSL-18.0.0 pyRFC3339-1.0 pycparser-2.14 python-augeas-0.5.0 pytz-2015.7 requests-2.20.0 requests-toolbelt-0.8.0 six-1.10.0 urllib3-1.24.1 zope.component-4.2.2 zope.event-4.1.0 zope.interface-4.1.3
apacheconftest runtests: commands[1] | /home/ubuntu/letsencrypt/certbot-apache/certbot_apache/tests/apache-conf-files/apache-conf-test --debian-modules

Reading package lists... 0%

Reading package lists... 100%

Reading package lists... Done


Building dependency tree... 0%

Building dependency tree... 0%

Building dependency tree... 50%

Building dependency tree... 50%

Building dependency tree... 62%

Building dependency tree... 95%

Building dependency tree       


Reading state information... 0%

Reading state information... 0%

Reading state information... Done

apache2 is already the newest version (2.4.34-1ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.

Reading package lists... 0%

Reading package lists... 100%

Reading package lists... Done


Building dependency tree... 0%

Building dependency tree... 0%

Building dependency tree... 50%

Building dependency tree... 50%

Building dependency tree       


Reading state information... 0%

Reading state information... 0%

Reading state information... Done

The following NEW packages will be installed:
  libapache2-mod-wsgi
0 upgraded, 1 newly installed, 0 to remove and 1 not upgraded.
Need to get 87.2 kB of archives.
After this operation, 273 kB of additional disk space will be used.

0% [Working]
            
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu cosmic/main amd64 libapache2-mod-wsgi amd64 4.5.17-1build1 [87.2 kB]

0% [1 libapache2-mod-wsgi 0 B/87.2 kB 0%]
                                         
100% [Working]
              
Fetched 87.2 kB in 0s (639 kB/s)
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
Selecting previously unselected package libapache2-mod-wsgi.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 59943 files and directories currently installed.)
Preparing to unpack .../libapache2-mod-wsgi_4.5.17-1build1_amd64.deb ...
Unpacking libapache2-mod-wsgi (4.5.17-1build1) ...
Setting up libapache2-mod-wsgi (4.5.17-1build1) ...
apache2_invoke: Enable module wsgi

Reading package lists... 0%

Reading package lists... 100%

Reading package lists... Done


Building dependency tree... 0%

Building dependency tree... 0%

Building dependency tree... 50%

Building dependency tree... 50%

Building dependency tree       


Reading state information... 0%

Reading state information... 0%

Reading state information... Done

E: Unable to locate package libapache2-mod-macro
enabling sslConsidering dependency setenvif for ssl:
Module setenvif already enabled
Considering dependency mime for ssl:
Module mime already enabled
Considering dependency socache_shmcb for ssl:
Module socache_shmcb already enabled
Module ssl already enabled
enabling rewriteModule rewrite already enabled
enabling macroEnabling module macro.
To activate the new configuration, you need to run:
  systemctl restart apache2
enabling wsgiModule wsgi already enabled
enabling deflateConsidering dependency filter for deflate:
Module filter already enabled
Module deflate already enabled
enabling userdirEnabling module userdir.
To activate the new configuration, you need to run:
  systemctl restart apache2
enabling versionERROR: Module version does not exist!
enabling mimeModule mime already enabled
enabling setenvifModule setenvif already enabled
testing 1626-1531.conf...passed
testing anarcat-1531.conf...passed
testing comment-continuations-2050.conf...passed
testing drupal-errordocument-arg-1724.conf...passed
testing drupal-htaccess-1531.conf...passed
testing escaped-space-arguments-2735.conf...passed
testing example-1755.conf...passed
testing example-ssl.conf...passed
testing example.conf...passed
testing finalize-1243.conf...failed
Root logging level set at -20 Saving debug log to /var/log/letsencrypt/letsencrypt.log Requested authenticator apache and installer apache Error while running apache2ctl configtest. Action 'configtest' failed. The Apache error log may have more information. AH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf: Cannot define multiple Listeners on the same IP:port Misconfigured PluginEntryPoint#apache: Error while running apache2ctl configtest. Action 'configtest' failed. The Apache error log may have more information. AH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf: Cannot define multiple Listeners on the same IP:port Traceback (most recent call last): File "/home/ubuntu/letsencrypt/certbot/plugins/disco.py", line 132, in prepare self._initialized.prepare() File "/home/ubuntu/letsencrypt/certbot-apache/certbot_apache/configurator.py", line 235, in prepare self.config_test() File "/home/ubuntu/letsencrypt/certbot-apache/certbot_apache/configurator.py", line 2213, in config_test raise errors.MisconfigurationError(str(err)) MisconfigurationError: Error while running apache2ctl configtest. Action 'configtest' failed. The Apache error log may have more information. AH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf: Cannot define multiple Listeners on the same IP:port Single candidate plugin: 1626-1531.conf README.modules anarcat-1531.conf comment-continuations-2050.conf drupal-errordocument-arg-1724.conf drupal-htaccess-1531.conf escaped-space-arguments-2735.conf example-1755.conf example-ssl.conf example.conf finalize-1243.apache2.conf.txt finalize-1243.conf graphite-quote-1934.conf ipv6-1143.conf ipv6-1143b.conf ipv6-1143c.conf ipv6-1143d.conf missing-quote-1724.conf modmacro-1385.conf owncloud-1264.conf rewrite-quote-1960.conf roundcube-1222.conf section-continuations-2525.conf section-empty-continuations-2731.conf semacode-1598.conf sslrequire-wordlist-1827.htaccess two-blocks-one-line-1693.conf apache Description: Apache Web Server plugin Interfaces: IAuthenticator, IInstaller, IPlugin Entry point: apache = certbot_apache.entrypoint:ENTRYPOINT Initialized: <certbot_apache.override_debian.DebianConfigurator object at 0x7fe22316d850> Prep: Error while running apache2ctl configtest. Action 'configtest' failed. The Apache error log may have more information. AH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf: Cannot define multiple Listeners on the same IP:port Single candidate plugin: 1626-1531.conf README.modules anarcat-1531.conf comment-continuations-2050.conf drupal-errordocument-arg-1724.conf drupal-htaccess-1531.conf escaped-space-arguments-2735.conf example-1755.conf example-ssl.conf example.conf finalize-1243.apache2.conf.txt finalize-1243.conf graphite-quote-1934.conf ipv6-1143.conf ipv6-1143b.conf ipv6-1143c.conf ipv6-1143d.conf missing-quote-1724.conf modmacro-1385.conf owncloud-1264.conf rewrite-quote-1960.conf roundcube-1222.conf section-continuations-2525.conf section-empty-continuations-2731.conf semacode-1598.conf sslrequire-wordlist-1827.htaccess two-blocks-one-line-1693.conf apache Description: Apache Web Server plugin Interfaces: IAuthenticator, IInstaller, IPlugin Entry point: apache = certbot_apache.entrypoint:ENTRYPOINT Initialized: <certbot_apache.override_debian.DebianConfigurator object at 0x7fe22316d850> Prep: Error while running apache2ctl configtest. Action 'configtest' failed. The Apache error log may have more information. AH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf: Cannot define multiple Listeners on the same IP:port Selected authenticator None and installer None Could not choose appropriate plugin: The apache plugin is not working; there may be problems with your existing configuration. The error was: MisconfigurationError("Error while running apache2ctl configtest.\nAction 'configtest' failed.\nThe Apache error log may have more information.\n\nAH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf:\nCannot define multiple Listeners on the same IP:port\n",) Exiting abnormally: Traceback (most recent call last): File "/home/ubuntu/letsencrypt/.tox/apacheconftest/bin/certbot", line 11, in <module> load_entry_point('certbot', 'console_scripts', 'certbot')() File "/home/ubuntu/letsencrypt/certbot/main.py", line 1365, in main return config.func(config, plugins) File "/home/ubuntu/letsencrypt/certbot/main.py", line 1229, in certonly installer, auth = plug_sel.choose_configurator_plugins(config, plugins, "certonly") File "/home/ubuntu/letsencrypt/certbot/plugins/selection.py", line 237, in choose_configurator_plugins diagnose_configurator_problem("authenticator", req_auth, plugins) File "/home/ubuntu/letsencrypt/certbot/plugins/selection.py", line 341, in diagnose_configurator_problem raise errors.PluginSelectionError(msg) PluginSelectionError: The apache plugin is not working; there may be problems with your existing configuration. The error was: MisconfigurationError("Error while running apache2ctl configtest.\nAction 'configtest' failed.\nThe Apache error log may have more information.\n\nAH00526: Syntax error on line 3 of /etc/apache2/sites-enabled/finalize-1243.conf:\nCannot define multiple Listeners on the same IP:port\n",) Please see the logfiles in /var/log/letsencrypt for more details.


testing graphite-quote-1934.conf...passed
testing ipv6-1143.conf...passed
testing ipv6-1143b.conf...passed
testing ipv6-1143c.conf...passed
testing ipv6-1143d.conf...passed
testing missing-quote-1724.conf...passed
testing modmacro-1385.conf...passed
testing owncloud-1264.conf...passed
testing rewrite-quote-1960.conf...passed
testing roundcube-1222.conf...passed
testing section-continuations-2525.conf...passed
testing section-empty-continuations-2731.conf...passed
testing semacode-1598.conf...passed
testing two-blocks-one-line-1693.conf...passed
ERROR: InvocationError: '/home/ubuntu/letsencrypt/certbot-apache/certbot_apache/tests/apache-conf-files/apache-conf-test --debian-modules'
___________________________________ summary ____________________________________
ERROR:   apacheconftest: commands failed
```

@joohoi, I don't think this is the highest priority Apache thing for you to look at, but if you get a chance, seeing if you can diagnose/fix the problems here would be great.

**Comment by bmw** [comment]
If this is resolved in a way that tests on Ubuntu 18.04 & 18.10 work, we need to update the release instructions to reflect this.

I'm happy to help with this when we get to that point.

**Comment by bmw** [comment]
@joohoi, this would be another useful thing for you to work on this week if you have the time. Maybe not the highest priority, but it'd be good to make sure there isn't a bug here and making tests pass on these OSes allows us to run Apache test farm tests on them which should help reduce the chance of regressions.
